lomap/graphgen.py

```diff
@@ -332,12 +362,16 @@ def minimizeEdges(self):
             if len(subgraph.edges()) > 2:   # Graphs must have at least 3 edges to be minimzed
 
                 for edge in weightsList:
-
-                    subgraph.remove_edge(edge[0], edge[1])
-
-                    if self.checkConstraints(subgraph, numberOfComponents) == False:
-                        subgraph.add_edge(edge[0], edge[1], similarity = edge[2], strict_flag = True)
-
+                    #sliu 07/17 add radial option here to avoid removing edges connect to the lead compound
```

Comment:
Yes, explain the logic more. For the option always applied question, actually, it'll only turn on when the lead_index is not None. If the radial option is not enabled, the lead_index will be None (default). And if the radial option is on, then the code will try to calculate the lead_index depending on the  complete/biased(hub) option. If it successfully find the lead_index, then the lead_index be passed here.